heat_infoblox/object_manipulator.py

```diff
@@ -36,32 +36,68 @@ def get_member(self, member_name, return_fields=None, extattrs=None):
             'member', obj, return_fields, extattrs
         )
 
-    def create_member(self, name=None, platform='VNIOS',
-                      mgmt=None, lan1={}, lan2=None,
-                      nat_ip=None):
+    def create_member(self, name=None, platform='VNIOS', ha_pair=False,
+                      vip=None, mgmt=None, lan2=None, nat_ip=None,
+                      use_v4_vrrp=False,
+                      node1_ha=None, node2_ha=None,
+                      node1_lan1=None, node2_lan1=None,
+                      node2_mgmt=None, vrid=None, lan2_vrid=None):
         member_data = {'host_name': name, 'platform': platform}
         extra_data = {}
 
-        if lan1.get('ipv4', None):
-            extra_data['vip_setting'] = lan1['ipv4']
-        if lan1.get('ipv6', None):
-            extra_data['ipv6_setting'] = lan1['ipv6']
+        if ha_pair:
+            node1 = {}
+            if node1_ha and node1_ha.get('ipv4', None):
+                node1['ha_ip_address'] = node1_ha['ipv4']['address']
+            if node1_lan1 and node1_lan1.get('ipv4', None):
+                node1['mgmt_lan'] = node1_lan1['ipv4']['address']
+            if node1_lan1 and node1_lan1.get('ipv6', None):
+                node1['mgmt_lan'] = node1_lan1['ipv6']['virtual_ip']
+            node2 = {}
+            if node2_ha and node2_ha.get('ipv4', None):
+                node2['ha_ip_address'] = node2_ha['ipv4']['address']
+            if node2_lan1 and node2_lan1.get('ipv4', None):
+                node2['mgmt_lan'] = node2_lan1['ipv4']['address']
+            if node2_lan1 and node2_lan1.get('ipv6', None):
+                node2['mgmt_lan'] = node2_lan1['ipv6']['virtual_ip']
+            extra_data = {
+                'enable_ha': True,
+                'router_id': vrid,
+                'node_info': [
+                    {'lan_ha_port_setting': node1},
+                    {'lan_ha_port_setting': node2}
+                    ]
+                }
+
+        if vip and vip.get('ipv4', None):
+                extra_data['vip_setting'] = vip['ipv4']
+        if vip and vip.get('ipv6', None):
+                extra_data['ipv6_setting'] = vip['ipv6']
         if nat_ip:
             extra_data['nat_setting'] = {
                 'enabled': True,
                 'external_virtual_ip': nat_ip
             }
 
         if mgmt and mgmt.get('ipv4', None):
-            extra_data['node_info'] = [{"mgmt_network_setting": mgmt['ipv4']}]
             extra_data['mgmt_port_setting'] = {"enabled": True}
+            if ha_pair:
+                extra_data['node_info'][0] = {
+                    "mgmt_network_setting": mgmt['ipv4']}
+                extra_data['node_info'][1] = {
+                    "mgmt_network_setting": node2_mgmt['ipv4']}
```

Comment:
node2_mgmt['ipv4'] exists for sure?

Overall this huge if else statements in this method seem a bit fragile.
I recommend to enhance more checks or refactor.